explainability-service/src/main/java/org/kie/trustyai/service/payloads/BaseExplanationRequest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.trustyai.service.payloads;
+
+public class BaseExplanationRequest {
```

Comment:
Overall these classes look quite similar to what we did in the past with Kogito integration ( https://github.com/kiegroup/kogito-apps/tree/main/explainability/explainability-api/src/main/java/org/kie/kogito/explainability/api ).
I don't think it makes sense to refactor Kogito but it might be interesting to consider if we want to move this "layer" of classes inside the `-core` or in some other module. 
For example it might be useful also for the interface to support different algorithms.

@ruivieira 
Wdyt?

Reply:
at this stage I've kept the current req/resp API as minimal as possible, I think it makes sense to consider that, perhaps for a subsequent enhancement PR.

Reply:
I've created https://issues.redhat.com/browse/RHODS-8820 to track this.